ibis/backends/base/__init__.py

```diff
@@ -713,6 +715,10 @@ def register_options(cls) -> None:
             except ValueError as e:
                 raise exc.BackendConfigurationNotRegistered(backend_name) from e
 
+    @abc.abstractmethod
+    def _register_udfs(self, expr: ir.Expr):
```

Comment:
An `abstractmethod` requires a subclass to implement it, and raises an error if it's unimplemented. We might instead raise an appropriate `NotImplementedError` (or equivalent) in the base implementation, rather than forcing all subclasses (even those not supporting UDFs) to implement this.

Reply:
Remove the abc and made methods raise `NotImplementedError`.